src/app/pages/products-list/card/card.component.ts

```diff
@@ -11,22 +11,25 @@ import {productsMock} from '../../../shared/products/products.mock';
     styleUrls: ['./card.component.css'],
 })
 export class CardComponent {
-    readonly product = productsMock[0];
-    // user: {name: string} = {name: 'Alex'};
-    // user = user;
+    @Input() product!: Product | null;
+    @Output() buyProduct = new EventEmitter<Product>();
 
-    // constructor() {
-    //     this.user = null;
-    // }
+    // readonly product = productsMock[0];
 
     onProductBuy(event: Event) {
         event.stopPropagation();
 
+        if (this.product) {
+            this.buyProduct.emit(this.product);
+        }
+
         // eslint-disable-next-line no-console
         console.log('Buy product');
     }
 
     isStarActive(starIndex: number): boolean {
-        return this.product.rating >= starIndex;
+        return (
+            this.product !== null && this.product !== undefined && this.product.rating >= starIndex
```

Comment:
А так можно было бы?
```suggestion
            !!this.product && this.product.rating >= starIndex
```